### PR TITLE
Logs for back end

### PR DESCRIPTION
**Pull request by DomMarq**

### CODE REVIEW

**Comment by DomMarq** [comment]
I have no idea why there are so many commits here. Most of them were merged in my last PR...

File: adapters/airflow_adapter.py [hunk 1]
```diff
@@ -142,17 +142,17 @@ def get_release_types(self):
...
-    types = set()
-    for release in releases:
-      branches.add(releases[release].branch)
-      types.add(releases[release].release_type)
-
-      with self._lock:
-        self._branches = list(branches)
-        self._release_types = list(types)
-        self._cache_last_updated = datetime.now()
+    else:
```
Comment (sebastienvas) [inline adapters/airflow_adapter.py]: why did you add this back ?

File: adapters/from_sql.py [hunk 2]
```diff
@@ -61,7 +60,8 @@ def read_releases(release_data, airflow_db):
         # it seems that some xcom dicts do not have this value for some reason?
         release.release_type = xcom_dict[xcomKeys.RELEASE_TYPE]
       except KeyError:
-        _, release.release_type = parse_dag_id(item.dag_id)
+        # old releases don't have release types, so this is a fallback
+        release.release_type, _ = parse_dag_id(item.dag_id)
```
Comment (sebastienvas) [inline adapters/from_sql.py:64]: is this expected that the order of arguments is changed ?

Comment (sebastienvas) [reply adapters/from_sql.py:64]: I guess that was the bug.

File: adapters/gcs_access.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import logging
+import os
+import lib.cloudstorage as gcs
+from resources.release_id_parser import release_id_parser
+
+def read_file(bucket_name, release_id, task_name):
```
Comment (sebastienvas) [inline adapters/gcs_access.py]: please move this to the airflow adapter as it only make sense for airflow.

File: adapters/to_sql.py [hunk 4]
```diff
@@ -45,14 +46,14 @@ def to_sql_release(dag_id, execution_date):
   # construct query
   sql_query = 'SELECT dag_id, execution_date FROM dag_run'
   sql_query += ' WHERE dag_id = "' + dag_id + '"'
-  sql_query += ' AND execution_date = "' + str(execution_date) + '"'
+  sql_query += ' AND execution_date LIKE "%' + str(execution_date) + '%"'
```
Comment (sebastienvas) [inline adapters/to_sql.py]: Why are we changing that ?

File: config [hunk 5]
```diff
@@ -0,0 +1,51 @@
+[core]
```
Comment (sebastienvas) [inline config]: PLease remove this file from your PR.

File: lib/cloudstorage/__init__.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# Copyright 2014 Google Inc. All Rights Reserved.
```
Comment (sebastienvas) [inline lib/cloudstorage/__init__.py:1]: Do we need this lib ?

File: adapters/airflow_adapter.py (hunk 1 above)
Comment (DomMarq) [reply adapters/airflow_adapter.py]: Must have snuck into the code in one of my merges. I'll get rid of it.

File: adapters/to_sql.py (hunk 4 above)
Comment (DomMarq) [reply adapters/to_sql.py]: I noticed that there were some releases, most of which were triggered manually, that were not showing up on the dashboard. I figured out that this is because the execution dates contained nanoseconds, which was not originally compensated for. By changing it to like, it ensures that the execution dates can contain nanoseconds, but do not have to. The fix actually caused about 35 more releases to be available on the dashboard.

File: lib/cloudstorage/__init__.py (hunk 6 above)
Comment (DomMarq) [reply lib/cloudstorage/__init__.py:1]: Unfortunately, yes I believe we do. It is what I am using to make calls to the GCS bucket, and it was the only one I tried that I could actually get to work without too much hassle.

File: adapters/airflow_adapter.py [hunk 7]
```diff
@@ -139,6 +141,25 @@ def get_release_types(self):
...
+      bucket_name: str
+      release_id: str
+      task_name: str
+
+    Returns:
+      Structured log text
+    """
+    dag_id, execution_date = release_id_parser(release_id)
+    execution_date = str(execution_date).replace(' ', 'T')  # put into same format as gcs bucket
+    filename = '/' + bucket_name + '/logs/' + dag_id + '/' + task_name + '/' + str(execution_date) + '/' + '1.log'
```
Comment (sebastienvas) [inline adapters/airflow_adapter.py]: And use log_file instead of 1.log
```python
os.path.join(os.path.sep, bucket_name , 'logs' , dag_id , task_name, execution_date,  log_file))
```

File: adapters/airflow_adapter.py [hunk 8]
```diff
@@ -139,6 +141,25 @@ def get_release_types(self):
     with self._lock:
       return self._release_types
 
+  def get_logs(self, bucket_name, release_id, task_name):
```
Comment (sebastienvas) [inline adapters/airflow_adapter.py]: add a log_file="1.log" default argument.

File: adapters/to_sql.py (hunk 4 above)
Comment (sebastienvas) [reply adapters/to_sql.py]: ok. Could you please briefly document that in the code?

Comment (DomMarq) [reply adapters/to_sql.py]: Done

**Review by sebastienvas — APPROVED** [review]
